### PR TITLE
[NU-1068] ModelClassLoader - ability to specify working directory for the tests purpose

### PR DESCRIPTION
**Pull request by arkadius**
## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/factory/LocalProcessingTypeDataStateFactory.scala [hunk 1]
```diff
@@ -22,7 +24,8 @@ class LocalProcessingTypeDataStateFactory(
   override def create(
       designerConfig: ConfigWithUnresolvedVersion,
       getDeploymentManagerDependencies: ProcessingType => DeploymentManagerDependencies,
-      additionalUIConfigProvider: AdditionalUIConfigProvider
+      additionalUIConfigProvider: AdditionalUIConfigProvider,
+      workingDirectoryOpt: Option[Path]
```
Comment (lciolecki) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/factory/LocalProcessingTypeDataStateFactory.scala:28]: Why Opt? I haven't seen much use for this kind of standard.

File: extensions-api/src/main/scala/pl/touk/nussknacker/engine/ProcessingTypeConfig.scala [hunk 2]
```diff
@@ -1,14 +1,10 @@
...
 import com.typesafe.config.Config
-import net.ceedubs.ficus.readers.ValueReader
-
-import java.io.File
-import java.net.URL
 
 case class ProcessingTypeConfig(
     deploymentManagerType: String,
-    classPath: List[URL],
+    classPath: List[String],
```
Comment (lciolecki) [inline extensions-api/src/main/scala/pl/touk/nussknacker/engine/ProcessingTypeConfig.scala:7]: Isn't it an API's class? Maybe we should put down an entry in MigrationGuide? Why we change URL to String? At the end you convert these strings to URI..

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/factory/LocalProcessingTypeDataStateFactory.scala (hunk 1 above)
Comment (arkadius) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/factory/LocalProcessingTypeDataStateFactory.scala:28]: `find -name *.scala -exec grep 'Opt: ' {} \; | wc -l` returned 127 entries - it is quite common. I've seen three approaches in scala:
1. using `Opt` suffix
2. using `maybe` prefix
3. skipping Opt/maybe (I assume that this one you propose)
 
1. and 2. approach has the benefit that whey you do map on this field you don't have to think how to call the mapped value (`workingDirectoryOpt.map { workingDirectory=>`). How do you name the mapped values in the proposed approach?

Comment (lciolecki) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/factory/LocalProcessingTypeDataStateFactory.scala:28]: So we can make agreement that in the future Option value we will named with postfix Opt? It's fine by me.

**Review by lciolecki — APPROVED** [review]

**Review by lukasz-bigorajski — APPROVED** [review]

**Pull request merged**